### PR TITLE
Use cb domain to sign preconfer api

### PR DESCRIPTION
**Pull request by Akagi201**
Change signature verification domain to commit-boost in elect-preconfer API.

## Summary by CodeRabbit

- **New Features**
	- Updated signature verification process for election requests to enhance security.
	- Introduced new functions for computing and verifying signed commit boost messages.

- **Documentation**
	- Updated comments and documentation to reflect new requirements for validating election requests.

### CODE REVIEW

**Comment by Akagi201** [comment]
@coderabbitai review

**Review by zsluedem — APPROVED** [review]
LGTM

**Review by chirag-bgh — APPROVED** [review]

File: crates/utils/src/signing.rs [hunk 1]
```diff
@@ -39,6 +42,18 @@ pub fn verify_signed_builder_message<T: HashTreeRoot>(
...
+    message: &mut T,
+    signature: &BlsSignature,
+    public_key: &BlsPublicKey,
+    genesis_validators_root: Option<Root>,
+    context: &Context,
+) -> Result<(), Error> {
+    let domain = compute_commit_boost_domain(context, genesis_validators_root)?;
+    verify_signed_data(message, signature, public_key, domain)?;
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (Akagi201) [reply crates/utils/src/signing.rs]: ignore

File: crates/utils/src/signing.rs [hunk 2]
```diff
@@ -65,3 +80,19 @@ pub fn compute_builder_domain(context: &Context) -> Result<Domain, Error> {
...
+pub fn compute_custom_domain(domain_mask: &[u8; 4], genesis_validators_root: Option<Root>, context: &Context) -> Result<Domain, Error> {
+    let fork_version = context.genesis_fork_version;
+    let genesis_validators_root = genesis_validators_root.unwrap_or_default();
+    let fork_data_root = compute_fork_data_root(fork_version, genesis_validators_root)?;
+
+    let mut domain = Domain::default();
+    domain[..4].copy_from_slice(domain_mask);
+    domain[4..].copy_from_slice(&fork_data_root[..28]);
+    Ok(domain)
+}
```
[the remark this answers is not in the document]
Comment (Akagi201) [reply crates/utils/src/signing.rs]: solved

**Comment by Akagi201** [comment]
@coderabbitai review

**Pull request merged**